packages/client/src/clients/slack/index.ts

```diff
@@ -33,20 +43,40 @@ class SlackClient {
     this.instance = instance;
   }
 
-  async getChannels(params: GetSlackChannelsInput) {
+  async authCheck({ tenant, knockChannelId }: AuthCheckInput) {
     const result = await this.instance.client().makeRequest({
       method: "GET",
-      url: `/v1/providers/slack/channels`,
+      url: `/v1/providers/slack/${knockChannelId}/auth_check`,
       params: {
         access_token_object: {
-          object_id: params.tenant,
+          object_id: tenant,
+          collection: TENANT_COLLECTION,
+        },
+        channel_id: knockChannelId,
+      },
+    });
+
+    return this.handleResponse(result);
+  }
+
+  async getChannels({
+    knockChannelId,
+    tenant,
+    connectionsObject,
+  }: GetSlackChannelsInput) {
+    const result = await this.instance.client().makeRequest({
+      method: "GET",
+      url: `/v1/providers/slack/${knockChannelId}/channels`,
+      params: {
+        access_token_object: {
+          object_id: tenant,
           collection: TENANT_COLLECTION,
         },
         connections_object: {
```

Comment:
Nit: Can probably just pass in `connections_object` instead of reassigning each property

Reply:
can't annoyingly because of `objectId` -> `object_id`